Assets/Scripts/UI/Windows/UploadMapPopUp.cs

```diff
@@ -0,0 +1,204 @@
+                httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + token.AccessToken);
+            }
+            catch (Exception e)
+            {
+                onError("Login failed, please check connection and login data. Exception is : \n"+e.Message);
+                return;
+            }
+
+            httpClient.DefaultRequestHeaders.Add("Accept","application/json");
+            try
+            {
+                HttpResponseMessage res = httpClient.PostAsync(BaseUrlInputField.text + "/maps/upload", form).Result;
+                try
+                {
+                    res.EnsureSuccessStatusCode();
+                }
+                catch (Exception e)
+                {
+                    onError(
+                        "JAVA-API complains: \n" + res.Content.ReadAsStringAsync().Result + "\n\nException is: \n" + e.Message);
+                }
+            }
+            catch (Exception e)
+            {
+                onError("Sending failed. Maybe map is too big... Exception was: \n"+e.Message);
+            }
+
+            httpClient.Dispose();
+            form.Dispose();
+            GenericInfoPopup.ShowInfo("Uploaded Successful... Check the vault to see ;)");
+        }
+
+        private bool checkUserInput()
+        {
+            if (string.IsNullOrEmpty(MapFolderPathInputField.text))
+            {
+                onError("Please select a map to upload");
+                return false;
+            }
+            if (string.IsNullOrEmpty(BaseUrlInputField.text))
+            {
+                onError("Please enter an upload target e.g. https://api.faforever.com");
+                return false;
+            }
+            if (string.IsNullOrEmpty(UsernameInputField.text))
+            {
+                onError("Please enter a username");
+                return false;
+            }
+            if (string.IsNullOrEmpty(PasswordInputField.text))
+            {
+                onError("Please enter a password");
+                return false;
+            }
+
+            return true;
+        }
+
+        private void onError(string errorMessage)
+        {
+            ErrorText.enabled = true;
+            var error = ErrorText.text = errorMessage;
+            Debug.LogError(error);
+        }
+
+        private IAuthorizationState GetAccessTokenFromOwnAuthSvr()
+        {
+            var server = new AuthorizationServerDescription
+            {
+                TokenEndpoint = new Uri(BaseUrlInputField.text+"/oauth/token"),
+                ProtocolVersion = ProtocolVersion.V20
+            };
+
+            var client = new UserAgentClient(server)
+            {
+                //Those are the credentials for Downlords-FAF-Client
+                ClientIdentifier = "3bc8282c-7730-11e5-8bcf-feff819cdc9f",
+                ClientCredentialApplicator = ClientCredentialApplicator.PostParameter("6035bd78-7730-11e5-8bcf-feff819cdc9f")
```

Comment:
Should not be hardcoded
And even more, should **not** use the Downlord FAF Client credentials but rather be registered as a new oauth app

Reply:
there is no setting file so there is no other option then hard code

Reply:
Wrong, the other option is : make the setting file.
Don't mark this as resolved, this is *not* resolved.